### ISSUE TITLE
support symfony3 or not？

### ISSUE THREAD

**Issue by benyang2015** [orocrm/platform#545]

**Comment by RocKordier** [comment]
@DimaSoroka has answered this already on orocrm forum  a year ago :)
https://www.orocrm.com/forums/topic/roadmap-regarding-upgrade-of-symfony#post-9422

**Comment by sadortun** [comment]
Oh, @RocKordier  This is in a distant future ! It's expected to be in November 2017 ! 

![image](https://cloud.githubusercontent.com/assets/794449/19223297/a5baafa8-8e3a-11e6-9230-32e073525dd6.png)

**Comment by artml** [comment]
Yes, next year we will start looking at migration to Symfony 3. More likely the second half of the year.

**Comment by stephanvierkant** [comment]
I just started to look into this project and this issue concerns me. I don't think this strategy (only use LTS releases) is a good idea:

- We have to wait at least a year before using other libraries/bundles that use Symfony 3. I'm developing many Symfony projects and almost all popular packages are already compatible with Symfony 3. Sticking to Symfony 2.8 means that new releases of those packages can't be used in a project that rely on orocrm.
- This strategy ignores the fact that Symfony isn't the only dependency that can contain performance issues, security issues or compatibility problems. Orocrm/platform has many other dependencies. It's not unlikely that one of those packages should be updated (to add PHP 7 support, fix security issues, etc.) in the future, but a new release of one of these packages require Symfony 3.x.
- If LTS is important because users want a stable, secure application, why does it require PHP 5.5.9?! PHP 5.5 is end of life for months now (even no security patches!) and 5.5.9 isn't even the latest version in the branch. [PHP 5.5.10 was released on 06 Mar 2014](http://www.php.net/ChangeLog-5.php#5.5.10)
- Upgrading from Symfony 2.8 to Symfony 3.0 isn't that hard. Upgrading from 2.8 to Symfony 3.4 (skipping 4 releases) is. Upgrading Symfony every 2,5 years isn't that _agile_.

My suggestion:
- Keep ORO CRM (master) up to date with the latest stable release of Symfony. That means upgrading to Symfony 3.1 before Symfony 3.0 is end of life, upgrading to 3.2 before 3.1.... you get the point.
- Think of a better strategy to release stable (LTS?) versions of ORO CRM without looking at just one of the dependencies.

**Comment by mbessolov** [comment]
@stephanvierkant 

PHP 5.5 is not a requirement - it is just a minimum supported version  of PHP. You may very well run OroCRM and OroPlatform on PHP 7.x which is fully supported as well.

And in regards to the minimum versions in general - we have certain obligations to our development community and customers in regards to the support of minimal versions of Symfony, PHP and MySQL that we need to honor. Not every community developer would have resources and capacity to keep up with the upgrades of the platform by producing a new version of their extension(s) if every time we were bumping up the Symfony version. And customers may have their own customizations, that would require rework if Symfony or another significant dependency version were changed.

Sooner or later the upgrade will happen anyway. And if you want to help make it sooner - we are accepting community contributions, and you are welcome to start the community initiative on upgrading the existing codebase to make it fully compatible with Symfony 3 in a separate branch. Symfony provides a number of tools to perform code analysis and produce the list of incompatibilities - which could be a good starting point to start this work on.

**Comment by stephanvierkant** [comment]
@mbessolov Thanks for your reply. Good to know PHP 7 is supported.

@artml's and Dima Soroka's ([this one](https://www.orocrm.com/forums/topic/roadmap-regarding-upgrade-of-symfony/#post-9422)) comments are very clear: we will start looking at Symfony 3 in the future (end 2017).

You are now saying community contributions are welcome to make it compatible with Symfony 3. Does that mean there will be a release that supports Symfony 3.0/3.1/3.2/3.3? Or do you really want to wait for a LTS version?

I really love Symfony because it's [Backwards Compatibility Promise](http://symfony.com/doc/current/contributing/code/bc.html) and clear [The Release Process](http://symfony.com/doc/current/contributing/community/releases.html). Do you have any documentation about your release process?

**Comment by mbessolov** [comment]
Our release process is documented in the [OroCRM Community Guide ](https://www.orocrm.com/documentation/index/current/community/release-process)

At this time our heads are down into preparation of the [next OroCRM release](https://www.orocrm.com/orocrm-and-platform-roadmap). And our backlog for the first half of 2017 is [already filled with feature requests](https://www.orocrm.com/orocrm-and-platform-roadmap/future-roadmap) from the OroCRM users. And the migration to Symfony 3.x will require some work and cannot be done overnight. So yes, there is an opportunity for the community to contribute. 

The next Symfony LTS release (3.4) is [scheduled for November 2017](http://symfony.com/roadmap?version=3.4). If the community wants to help and speed up the process, for example to help us achieve full compatibility of OroCRM/OroPlatform with Symfony 3.4 by November 2017, so that, for example, we could publish our release the next day after Symfony 3.4 is announced  - that looks like quite a reasonable goal. If you want to achieve the Symfony 3.x compatibility even earlier than that - it would be extraordinary.

**Comment by videni** [comment]
it is very sad to know that this can only be done in the end of 2017.

**Comment by sadortun** [comment]
@videni You are more than welcome to submit a few PRs :) Maybe it could go faster :) !

**Comment by videni** [comment]
@sadortun, the problem is oro cummunity does not start to  symfony3 migration yet,  even though we submit PRs, they would'nt be meged.

**Comment by sadortun** [comment]
I think that most of the non-BC changes would be merged, since it can help the various repositories to move toward `3.x`, as you mentioned, if there is BC breaks thoses will have to wait ...

**Comment by possi** [comment]
As Symfony 3.4 LTS is here now, may we get a Roadmap for migration?

**Comment by mbessolov** [comment]
We've already checked what it would take to upgrade, and it seems that if we set a few developers aside to do that, we may be able to finish the migration to 3.4 LTS in the middle of the next year (2018). But considering general lack of interest from the community, and a lot more voices asking about new features vs Symfony upgrade, we are not commiting to that yet.

**Comment by alecslupu** [comment]
@mbessolov, I have tried to upgrade to symfony 3.0.9, and it seems the bottleneck is sylius/flow-bundle which is abandoned. I do not know what i could use to replace it. Maybe symfony-workflows ?

**Comment by alecslupu** [comment]
Also got those warnings: 

Package guzzle/guzzle is abandoned, you should avoid using it. Use guzzlehttp/guzzle instead.
Package phpoffice/phpexcel is abandoned, you should avoid using it. Use phpoffice/phpspreadsheet instead.

**Comment by deki** [comment]
@mbessolov I think there is some interest as the current version has end of support for bug fixes in Nov 2018 (see https://symfony.com/roadmap?version=2.8#checker). Will you update Oro by then?

I can understand the concerns, I wouldn't start implementing stuff with a technology that has end of life by end of the year.

**Comment by deki** [comment]
Great, exactly what I was looking for. I think you should resolve that issue here...

**Comment by borNfreee** [comment]
> OroPlatform 3.0 BETA already released.

But as I see it based not on symfony 3.4, but on 2.8 https://github.com/oroinc/platform/blob/3.0.0-beta/composer.json#L21

Could you please explain it? Probably I missed something

**Comment by anyt** [comment]
Symfony 3.4 will be required at the final 3.0.0 release, in beta our team made a lot of changes to remove deprecations, it’s the first step to upgrade.

**Comment by borNfreee** [comment]
https://oroinc.com/orocrm/blog/ready-version-3.0-oro-products

> As we upgraded all the Oro applications to Symfony 3.4, we’ll update our own version numbering scheme accordingly.

Please update the blogpost to avoid confusing. You are *going* to upgrade, not upgraded yet.

Anyway, thanks for starting this process!

**Comment by anyt** [comment]
I’ll send your request to technical writers. Thank you!
